### PR TITLE
Rename `TimestampCollector` to `AddressBookUpdater`

### PR DESCRIPTION
**Pull request by upbqdn**
This PR renames `TimestampCollector` to `AddressBookUpdater` because `TimestampCollector` is not used to only collect timestamps.

Resolves the conversation at https://github.com/ZcashFoundation/zebra/pull/2974#discussion_r739467360.

Anyone can review, but @jvff and @teor2345 have most of the context.

### CODE REVIEW

**Comment by upbqdn** [comment]
The tests fail because https://github.com/ZcashFoundation/zebra/pull/2974 is not finished.

**Comment by teor2345** [comment]
I'm going to mark this PR as draft, because it depends on a draft PR.
Feel free to switch it back when it's ready to merge.

File: zebra-network/src/peer/handshake.rs [hunk 1]
```diff
@@ -409,7 +409,7 @@ where
             let (tx, _) = broadcast::channel(100);
             tx
         });
-        let timestamp_collector = self.timestamp_collector.unwrap_or_else(|| {
+        let address_book_updater = self.address_book_updater.unwrap_or_else(|| {
             // No timestamp collector was passed, so create a stub channel.
             // Dropping the receiver means sends will fail, but we don't care.
```
Comment (jvff) [inline zebra-network/src/peer/handshake.rs]: Maybe update this comment as well?

```suggestion
            // No `AddressBookUpdater` for timestamp collection was passed, so create a stub
            // channel. Dropping the receiver means sends will fail, but we don't care.
```

**Review by jvff — APPROVED** [review]
Looks good. Just one minor suggestion to update a comment for clarity.

**Comment by upbqdn** [comment]
Just to clarify, and maybe help with the review: I `ripgrep`ed the codebase, and there should be no occurrences of `TimestampCollector` or `timestamp_collector` anymore.

**Comment by upbqdn** [comment]
This is ready to be merged to `add-peers-to-address-book`.

**Review by jvff — APPROVED** [review]

**Pull request merged**